params/config.go

```diff
@@ -436,10 +436,10 @@ func (c *CliqueConfig) String() string {
 
 // OptimismConfig is the optimism config.
 type OptimismConfig struct {
-	EIP1559Elasticity        uint64 `json:"eip1559Elasticity"`
-	EIP1559Denominator       uint64 `json:"eip1559Denominator"`
-	EIP1559DenominatorCanyon uint64 `json:"eip1559DenominatorCanyon"`
-	EnableL2Blob             bool   `json:"enable_l2_blob"`
+	EIP1559Elasticity        uint64  `json:"eip1559Elasticity"`
+	EIP1559Denominator       uint64  `json:"eip1559Denominator"`
+	EIP1559DenominatorCanyon uint64  `json:"eip1559DenominatorCanyon"`
+	L2BlobTime               *uint64 `json:"l2BlobTime"`
```

Comment:
Should we use ChainConfig rather than OptimismConfig?  ChainConfig seems to be single source to setup the time-based fork rule.

Reply:
I'd like to scope our changes within `OptimismConfig` for better maintainability, since if this code is to be merged into upstream, it's not going to be call L2Blob HF, and we need to replace L2Blob HF with the actual XXX HF anyway.

Reply:
Then we will also replace `L2BlobTime` here with he actual upgrade name.  That means, we need to replace the config anyway. The reason to put it in `OptimismConfig` also improve maintainability.

For example, the json flag should be marked with `omitempty`, which should set the value as nil (disable) if not specified. This issue happens because OptimismConfig is designed for non-nil values.  If we use ChainConfig as the source, then it will be easier to identify the issue of missing `omitempty`.

Reply:
Good catch for `omitempty`, but another reason for putting `L2BlobTime` inside `OptimismConfig` is that we only want to disallow blob tx when Optimism is enabled(OptimismConfig is not nil).

Reply:
> Good catch for `omitempty`, but another reason for putting `L2BlobTime` inside `OptimismConfig` is that we only want to disallow blob tx when Optimism is enabled(OptimismConfig is not nil).

According to current logic, if OptimismConfig is nil, then we will also disable blob tx, right? I am a bit confused here.

Reply:
If OptimismConfig is nil, we'll not disable blob tx.

Reply:
```
// IsL2Blob returns whether l2 blob is enabled
func (c *ChainConfig) IsL2Blob(num *big.Int, time uint64) bool {
	return c.IsCancun(num, time) && c.Optimism != nil && isTimestampForked(c.Optimism.L2BlobTime, time)
}
```

Would the code here return false if c.Optimism is nil?

Reply:
Yes.

Reply:
> If OptimismConfig is nil, we'll not disable blob tx.

Then we will disable blob tx if OptimismConfig is nil, right?

Reply:
Nope, the filtering will only be applied if:

`OptimismConfig != nil && !L2Blob`

the filtering will not be applied if:

`OptimismConfig == nil || L2Blob`

Reply:
> Good catch for `omitempty`, but another reason for putting `L2BlobTime` inside `OptimismConfig` is that we only want to disallow blob tx when Optimism is enabled(OptimismConfig is not nil), the filtering is only for L2.

In fact, all other Optimism upgrades (e.g., `HoloceneTime`) depend on OptimismConfig not being nil, while they are set in `ChainConfig`, so the reasoning here is not sound to me. So I think it will be valuable to follow the upsteam convention and set the fork time in `ChainConfig`.

In the future, I will imagine if the L2BLOB is supported on upstream, it will be a fork time with a different name in `ChainConfig` so that we can just rename `L2BlobTime` rather than move the variable from `OptimismConfig`.

Reply:
Ok, updated.

Reply:
Ok, updated. It's a good move, spot another issue in monorepo, will change later.